File: PRACTICAS/PRACTICA_ACTUAL/RUBY/pkg/Civitas-0.0.1/lib/sorpresa_salircarcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_salircarcel < Sorpresa
    
    def initialize(n)
      @mazo = n
    end
    
    
  
    
    def aplicar_a_jugador(actual,todos)
      if actual>=0&&actual<todos.size
        informe(actual,todos)
        tiene_sorpresa_evitar_carcel=false
        for j in todos
          if(j.tiene_salvoconducto)
            tiene_sorpresa_evitar_carcel=true
          end
        end
        if !tiene_sorpresa_evitar_carcel
          todos[actual].obtener_salvoconducto(self)
          salir_del_mazo
        end
      end
    end
    
      def salir_del_mazo
        @mazo.inhabilitar_carta_especial(self)
      end
      
    def usada
        @mazo.habilitar_carta_especial(self)
    end
    
    
  end
end

File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/SorpresaSalirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author angelsolano
 */
public class SorpresaSalirCarcel extends Sorpresa{
    
    private MazoSorpresas mazo;
    
    SorpresaSalirCarcel(MazoSorpresas mazo, String texto){
        
        super(texto);
        this.mazo = mazo;

    }
    
  
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            Boolean tienenCarcel = false;
            for (int i = 0; i < todos.size() && !tienenCarcel; i++){
                if (i!=actual)
                    if (todos.get(i).tieneSalvoconducto())
                        tienenCarcel = true;
            }
            
            if(!tienenCarcel){
                    todos.get(actual).obtenerSalvoconducto(this);
                    salirDelMazo();
            }
            
            
        }
    }
    
    void salirDelMazo(){
            mazo.inhabilitarCartaEspecial(this);
    }
    
    void usada (){
            mazo.habilitarCartaEspecial(this);
    }
}
